plugins/pluginmanagers/lz-n.nix

```diff
@@ -0,0 +1,151 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib) mkOption;
+  inherit (helpers)
+    mkNullOrLuaFn'
+    nixvimTypes
+    keymaps
+    defaultNullOpts
+    ;
+
+  name = "lz-n";
+  originalName = "lz.n";
+
+  mkLazyLoadOption =
+    {
+      originalName ? "this plugin",
+    }:
+    let
+      pluginDefault = {
+        name = originalName;
+      };
+    in
+    mkOption {
+      description = "Lazy-load settings for ${originalName}.";
+      type =
+        with nixvimTypes;
+        let
+          nullStrToList = coercedTo str toList (listOf str);
```

Comment:
I'm _unsure_ a coerced type makes sense here, _ideally_ we want `plugins` to closely reflect the lz.n's config schema without too much magic on our end.